extensions/reg-core-3.3.md

```diff
@@ -0,0 +1,224 @@
+numerics:
+
+| No. | Label                             | Format                                                                   |
+| --- | --------------------------------- | ------------------------------------------------------------------------ |
+| 924 | `ERR_ACCOUNT_ALREADY_VERIFIED`    | `:<server> 924 <user_nickname> <accountname> :Account already verified`  |
+| 925 | `ERR_ACCOUNT_INVALID_VERIFY_CODE` | `:<server> 925 <user_nickname> <accountname> :Invalid verification code` |
+
+## The `REGCOMMANDS` `RPL_ISUPPORT` token
+
+Implementations which provide these commands SHOULD signal their availability using the `REGCOMMANDS`
+RPL_ISUPPORT (005) token.  The `REGCOMMANDS` token takes a comma-separated list of supported
+commands.  Other extensions MAY include additional commands in the `REGCOMMANDS` token than those
+documented here.
+
+A sample 005 reply indicating support for these commands is:
+
+    :irc.example.com 005 kaniini REGCOMMANDS=CREATE,VERIFY :are supported by this server
+
+## The `REGCALLBACKS` `RPL_ISUPPORT` token
+
+Implementations which require a verification token should list the types of verification callbacks
+they support, using the `REGCALLBACKS` RPL_ISUPPORT (005) token.  The `REGCALLBACKS` token takes
+a comma-separated list of supported commands.  In general, the `REGCALLBACKS` token is limited to
+URI schemes defined in the IANA URI schemes registry, per [RFC 4395][rfc4395].
+
+  [rfc3495]: https://tools.ietf.org/html/rfc4395
+
+In the event that verification is not required by the implementation, the `REGCALLBACKS` token should
+be an empty value.
+
+A sample 005 reply indicating support for verification callbacks is:
+
+    :irc.example.com 005 kaniini REGCALLBACKS=mailto,sms :are supported by this server
+
+A sample 005 reply indicating that no verification callback methods are supported is:
+
+    :irc.example.com 005 kaniini REGCALLBACKS= :are supported by this server
+
+The first callback type SHOULD be the implementation-defined default if default callback types
+are supported.
+
+## The `REGCREDTYPES` `RPL_ISUPPORT` token
+
+Implementations which provide support for this framework SHOULD specify the allowed credential
+types using the `REGCREDTYPES` RPL_ISUPPORT (005) token.  The `REGCREDTYPES` token takes a
+comma-separated list of supported commands.  Other extensions MAY include additional credential
+types than those listed here.
+
+The following credential types are defined:
+
+  * `passphrase`: an unencrypted passphrase which is sent to the server.
+  * `certfp`: the client certificate fingerprint as determined by the server if available.
+    The client MAY provide an empty credential (`*`) in this case.  The server SHOULD calculate
+    the credential using the client certificate fingerprint, if available.  If no client certificate
+    is presented, the server SHOULD NOT advertise the availability of this credential type.
```

Comment:
What hash algorithm should be used to give a fingerprint other than `*`? And what format?